### PR TITLE
Suppress `tar_terra_sprc()` warning

### PR DESCRIPTION
**Pull request by Aariq**
Addresses #92 by wrapping the write function in `withCallingHandlers()` to capture the uninformative warning `[rast] skipped sub-datasets ...`.  The warning originates from `terra::writeRaster()` when it is used to write a subdataset because the return value of `writeRaster()` is `rast(<file>)` and that file can't be correctly read by `rast()` if it has subdatasets with different projections.

This was difficult to add a test for since warnings from write functions apparently don't bubble up to the output of `tar_meta()`.  I added a test for a version of the write function instead.

### CODE REVIEW

File: tests/testthat/test-tar-terra-sprc.R [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
+    )
+    targets::tar_script({
+        elev_scale <- function(z = 1) {
+            terra::rast(
+                system.file(
+                    "ex",
+                    "elev.tif",
+                    package = "terra"
+                )
+            ) * z
```
Comment (Aariq) [inline tests/testthat/test-tar-terra-sprc.R:51]: Sorry, this was all just fixing indenting--there's no real changes here

**Comment by Aariq** [comment]
Warnings from `tar_format()` now show up in `tar_meta()` in the dev version of `targets`.  Can simplify this test eventually. https://github.com/ropensci/targets/discussions/1345#discussioncomment-10908585

File: NEWS.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 * Created utility function `set_window()` mostly for internal use within `tar_terra_tiles()`.
 * Removes the `iteration` argument from all `tar_*()` functions.  `iteration` now hard-coded as `"list"` since it is the only option that works (for now at least).
 * Added the `description` argument to all `tar_*()` functions which is passed to `tar_target()`.
+* Suppressed the warning "[rast] skipped sub-datasets" from `tar_terra_sprc()`, which is misleading in this context (#92, #104)
```
Comment (njtierney) [inline NEWS.md:9]:
```suggestion
* Suppressed the warning "[rast] skipped sub-datasets" from `tar_terra_sprc()`, which is misleading in this context (#92, #104).
```

**Review by njtierney — APPROVED** [review]
Looks good to me! I haven't used calling handlers before or https://rlang.r-lib.org/reference/cnd_muffle.html before, so that is cool to see!

**Pull request merged**